### PR TITLE
Do not use SSE flags on arm64 platform

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
SSE is not available on arm64, so the fact that we pass these flags causes compilation errors...

### CODE REVIEW

**Comment by SiarheiFedartsou** [comment]
@albarrentine please take a look when you have chance 🙏🏻

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,7 +11,10 @@ jobs:
   build_and_test:
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest]
+        os: 
+          - ubuntu-latest
+          - macos-15-large # x86_64
```
Comment (SiarheiFedartsou) [inline .github/workflows/test.yml:16]: it is a bit confusing, but it is indeed how GH Actions define arm64 and x86_64 images :)

**Comment by missinglink** [comment]
Hey @SiarheiFedartsou I just merged [this today in node-postal](https://github.com/openvenues/node-postal/pull/54/files) which required targeting `aarch64|arm64` since the ubu-arm image returned `aarch64` for `uname -m`

Do we possibly need to add that as an alias here too?

**Comment by SiarheiFedartsou** [comment]
> Hey @SiarheiFedartsou I just merged [this today in node-postal](https://github.com/openvenues/node-postal/pull/54/files) which required targeting `aarch64|arm64` since the ubu-arm image returned `aarch64` for `uname -m`
> 
> Do we possibly need to add that as an alias here too?

hm, indeed, I think so, I tested only on macOS :) 

I'll create a fix nearest days. Thanks for letting me know!

**Pull request merged**